packages/plugin-ext/src/common/plugin-protocol.ts

```diff
@@ -57,8 +58,18 @@ export interface PluginPackage {
     extensionPack?: string[];
 }
 export namespace PluginPackage {
+    function encodeFileURIComponent(value: string) {
+        return encodeURIComponent(value).replace(/%(2f|5c)/gi, "/")
+    }
+
     export function toPluginUrl(pck: PluginPackage, relativePath: string): string {
-        return `hostedPlugin/${getPluginId(pck)}/${encodeURIComponent(relativePath)}`;
+        const encodedRelativePath = encodeFileURIComponent(relativePath);
+        if (environment.electron.is()) {
```

Comment:
Could you elaborate why relative path does not work? There is an http endpoint which should process it? Does not it get hit? 

The thing is that we should not introduce a variant if there is a way just with the web tech.

Reply:
The endpoint indeed does not work, I don't think it gets hit because the relative `hostedPlugin/...` URL is resolved to a file:/// URL. If the Express server is available here it would probably need to be accessed via localhost.

We actually had a workaround where we would copy the icons to the correct location under `lib/hostedPlugin/...`, which worked when running with `theia start` but did not when everything was bundled up into the Electron .asar file.

Reply:
FWIW icons in the editor gutter do work and they seem to be using another URL (`http://example.org/file?file:///path/to/icon`) both in web and Electron. In Electron the host is localhost.

Reply:
Maybe we are better to fix it here: https://github.com/vrubezhny/theia/blob/e450add7771d7d8da3d2506e819e26555b8147f0/packages/plugin-ext/src/main/browser/plugin-shared-style.ts#L104-L105

We could then use [Endpoint](https://github.com/eclipse-theia/theia/blob/e5ab531832580186a75e6c6caa9ebbc674386fda/packages/core/src/browser/endpoint.ts#L38)  to create a proper URL which works for browser and electron in the same way.

Reply:
Thanks for the pointer! Done in https://github.com/eclipse-theia/theia/pull/7202/commits/af8fc766662b4205f794e5afa6858a73371bef1a. This now pretty much matches what happens with the gutter icons ([see here](https://github.com/eclipse-theia/theia/blob/8e60e099aee749872136372acb0ebb5aabff48f4/packages/plugin-ext/src/main/browser/text-editors-main.ts#L160)) and produces a similar URL that works in both browser and Electron.